### PR TITLE
Refactored code in compliance with Mercury framework PR #62

### PR DESCRIPTION
**Pull request by IlonaShishov**
Signed-off-by: Ilona Shishov <Ilona.Shishov@gmail.com>

### CODE REVIEW

File: coreless-ui/frontend/src/views/forms/customer-offer/CustomerOffer.js [hunk 1]
```diff
@@ -91,9 +91,11 @@ export default class CustomerOffer extends React.Component {
     const requestOptions = {
       method: 'POST',
       headers: { 'Content-Type': 'application/json' },
-      body: JSON.stringify({ procedure: { customerReference: this.state.customerFirstName } }),
+      body: JSON.stringify({
+        customerOfferProcedureInstanceRecord: { customerReference: this.state.customerFirstName },
```
Comment (ruromero) [inline coreless-ui/frontend/src/views/forms/customer-offer/CustomerOffer.js]: please format it better
```suggestion
        customerOfferProcedureInstanceRecord: { 
             customerReference: this.state.customerFirstName 
        },
```

File: coreless-ui/server/index.js [hunk 2]
```diff
@@ -18,12 +18,12 @@ var corsOptions = {
 };
 
 const PORT = process.env.PORT || 8080;
-const SERVICE_HOST = process.env.CO_SERVICE_HOST || '192.168.49.2';
-const SERVICE_PORT = process.env.CO_SERVICE_PORT || '31245';
-const PRP_SERVICE_HOST = process.env.PRP_SERVICE_HOST || '192.168.49.2';
-const PRP_SERVICE_PORT = process.env.PRP_SERVICE_PORT || '32041';
+const SERVICE_HOST = process.env.CO_SERVICE_HOST || 'customer-offer-camelk-rest-ishishov.apps.ocp-dev01.lab.eng.tlv2.redhat.com';
```
Comment (ruromero) [inline coreless-ui/server/index.js]: Don't hardcode ips or hostnames. Rely on environment variables only

File: coreless-ui/server/index.js [hunk 3]
```diff
@@ -18,12 +18,12 @@ var corsOptions = {
...
+const SERVICE_PORT = process.env.CO_SERVICE_PORT || '80';
+const PRP_SERVICE_HOST = process.env.PRP_SERVICE_HOST || 'party-routing-profile-camelk-rest-ishishov.apps.ocp-dev01.lab.eng.tlv2.redhat.com';
```
Comment (ruromero) [inline coreless-ui/server/index.js]: Same, this is a bad practice

File: coreless-ui/frontend/src/views/forms/party-routing-profile/PartyRoutingProfile.js [hunk 4]
```diff
@@ -18,31 +18,44 @@ export default class PartyRoutingProfile extends React.Component {
 
     this.state = {
       dataArray: [],
+      dataMap: new Map(),
       error: '',
     }
   }
   async componentDidMount() {
-    await fetch('/partyRoutingProfile')
+    await fetch('/getPartyRoutingProfileKeys')
```
Comment (ruromero) [inline coreless-ui/frontend/src/views/forms/party-routing-profile/PartyRoutingProfile.js]: can't you make it work asynchronously? I don't like the await that blocks the thread.

**Review by ruromero — CHANGES_REQUESTED** [review]

File: coreless-ui/frontend/src/views/forms/customer-offer/CustomerOffer.js (hunk 1 above)
Comment (IlonaShishov) [reply coreless-ui/frontend/src/views/forms/customer-offer/CustomerOffer.js]: fixed, thank you

File: coreless-ui/server/index.js (hunk 2 above)
Comment (IlonaShishov) [reply coreless-ui/server/index.js]: fixed, thank you

File: coreless-ui/server/index.js (hunk 3 above)
Comment (IlonaShishov) [reply coreless-ui/server/index.js]: fixed, thank you

File: coreless-ui/frontend/src/views/forms/party-routing-profile/PartyRoutingProfile.js (hunk 4 above)
Comment (IlonaShishov) [reply coreless-ui/frontend/src/views/forms/party-routing-profile/PartyRoutingProfile.js]: fixed, thank you

File: coreless-ui/frontend/src/views/forms/customer-offer-report/CustomerOfferReport.js [hunk 5]
```diff
@@ -12,69 +12,82 @@ import {
...
       method: 'PUT',
-      headers: { 'Content-Type': 'application/json' },
-      body: JSON.stringify({ procedure: { customerReference: item } }),
+      headers: { 'Content-Type': 'application/json', id: item.id },
+      body: JSON.stringify({
+        CustomerOfferProcedure: { CustomerReference: item.customerReference },
+      }),
     }
-    fetch('/UpdateCustomerOffer', requestOptions)
+    await fetch('/UpdateCustomerOffer', requestOptions)
```
Comment (ruromero) [inline coreless-ui/frontend/src/views/forms/customer-offer-report/CustomerOfferReport.js]: don't use await, process asynchronously

**Review by ruromero — CHANGES_REQUESTED** [review]

File: coreless-ui/frontend/src/views/forms/customer-offer-report/CustomerOfferReport.js (hunk 5 above)
Comment (IlonaShishov) [reply coreless-ui/frontend/src/views/forms/customer-offer-report/CustomerOfferReport.js]: fixed, thank you

**Review by ruromero — APPROVED** [review]
LGTM

**Pull request merged**